extern/extern.cmake

```diff
@@ -0,0 +1,50 @@
+### PHYSFS ###
+
+set(PHYSFS_BUILD_STATIC ON)
+set(PHYSFS_BUILD_SHARED OFF)
+set(PHYSFS_BUILD_TEST OFF)
+add_subdirectory("extern/physfs")
+
+add_library(extern::physfs INTERFACE IMPORTED)
+set_target_properties(extern::physfs PROPERTIES
+	INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_SOURCE_DIR}/extern/physfs/src
+	INTERFACE_LINK_LIBRARIES physfs-static
+)
```

Comment:
These lines above set up an "interface" library. CMake has the concept of public/interface/private. Public = needed for compilation of library and for use, interface = only needed for use, private = only needed for compilation. An interface library is a library that _ONLY_ has interface needs, good for passing along transitive dependencies.